File: crates/varde-code/tests/fixtures/swift/fixture.swift
// Entity extraction fixture for Swift: covers all 14 required kinds.
import Foundation

// Add adds two integers (public top-level function -> Function + Export).
public func add(a: Int, b: Int) -> Int {
    return a + b
}

// User is a public class -> Class + Export.
public class User {
    let name: String
    var age: Int

    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }

    func greet() -> String {
        return "hello " + name
    }
}

// Greeter is a protocol (Swift's abstraction mechanism) -> Interface.
protocol Greeter {
    func greet() -> String
}

// Top-level variables -> Variable.
let version = "1.0.0"
var flag = true

// GreeterImpl conforms to Greeter.
class GreeterImpl: Greeter {
    func greet() -> String {
        return "hi"
    }
}

func main() {
    let app = Application()
    // Route: Vapor-style app.get("/users") { ... }.
    app.get("/users") { req in
        let res = Response()
        // Response: res.send(...).
        res.send("ok")
    }
    app.run()
}

func compute(x: String, n: Int) -> String {
    let names = ["a", "b"]
    for name in names {
        if name.count > 0 {
            throw RuntimeError("bad name")
        }
    }
    var i = 0
    while i < n {
        i += 1
    }
    do {
        let result = try risky()
        return result
    } catch let err {
        return "caught"
    }
}

// Point is a struct (also class_declaration in this grammar).
struct Point {
    let x: Int
    let y: Int
}

File: crates/varde-code/tests/fixtures/kotlin/fixture.kt
// Entity extraction fixture for Kotlin: covers all 14 required kinds.
package com.example.app

import io.ktor.server.engine.embeddedServer
import io.ktor.server.netty.Netty
import io.ktor.server.response.respondText
import io.ktor.server.routing.get
import io.ktor.server.routing.routing

// Add adds two integers (public top-level function -> Function + Export).
public fun add(a: Int, b: Int): Int {
    return a + b
}

// User is a public data class -> Class + Export.
public data class User(val name: String, val age: Int)

// Greeter is an interface (Kotlin's abstraction mechanism) -> Interface.
interface Greeter {
    fun greet(): String
}

// Top-level properties -> Variable.
val version = "1.0.0"
const val maxRetries = 3
val isReady = true

// GreeterImpl implements Greeter; greet() is a method -> Function.
class GreeterImpl : Greeter {
    override fun greet(): String {
        val message = "hello " + version
        return message
    }
}

fun main() {
    val server = embeddedServer(Netty, port = 8080) {
        routing {
            // Route: Ktor-style get("/users") inside a routing block.
            get("/users") {
                // Response: call.respondText(...).
                call.respondText("users")
            }
            get("/health") {
                call.respondText("ok")
            }
        }
    }
    server.start(wait = true)
}

fun compute(x: String, n: Int): String {
    val names = listOf("a", "b")
    for (name in names) {
        if (name.length > 0) {
            throw IllegalArgumentException("bad name")
        }
    }
    var i = 0
    while (i < n) {
        i++
    }
    val result = try {
        x
    } catch (e: Exception) {
        "caught"
    }
    val fallback = result ?: "none"
    if (fallback.length > 0 && n > 1) {
        return fallback
    }
    return "empty"
}
